### PR TITLE
Input extensions

### PR DESCRIPTION
**Pull request by basvandriel**
Basic `dict` support for input extensions, to support automatic persisted queries in the future. See #1707.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #3296

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @bvandriel for the PR 👏

Get it here 👉 https://beta.strawberry.rocks/release/(next)
```

File: strawberry/schema/hash.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from hashlib import sha256
+
+
+def hash256(text: str) -> str:
+    return sha256(text.encode("utf-8")).hexdigest()
+
+
+def is_valid_hash256(text: str) -> bool:
+    return len(text) == 64
```
Comment (patrick91) [inline strawberry/schema/hash.py:9]: do we need these functions?

File: tests/cli/snapshots/unions_py38.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
-import strawberry
 from typing import Annotated, Union
 
+import strawberry
+
```
Comment (patrick91) [inline tests/cli/snapshots/unions_py38.py:4]: is this done by ruff? I think these files should be excluded 😊

File: tests/http/clients/base.py [hunk 3]
```diff
@@ -122,8 +129,9 @@ def _build_body(
         variables: Optional[Dict[str, object]] = None,
         files: Optional[Dict[str, BytesIO]] = None,
         method: Literal["get", "post"] = "post",
+        extensions: Optional[Dict[str, Any]] = None,
     ) -> Optional[Dict[str, object]]:
-        if query is None:
+        if query is None and extensions is None:
             assert files is None
             assert variables is None
```
Comment (patrick91) [inline tests/http/clients/base.py:136]: is this additional check necessary? 🤔

Comment (basvandriel) [reply tests/http/clients/base.py:136]: It's not. Was a part of the previous work I overseen

File: strawberry/schema/hash.py (hunk 1 above)
Comment (basvandriel) [reply strawberry/schema/hash.py]: Not yet :)

**Comment by basvandriel** [comment]
Can someone take a look at this PR? Tests in `tests/websockets/test_graphql_transport_ws.py` seem to be failing in the CI which I can't replicate locally. cc @patrick91

**Pull request closed without merging**